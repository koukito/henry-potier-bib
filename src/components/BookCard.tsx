import React from 'react';

import { Button, Card } from 'react-bootstrap';

import { Book } from '../models/books.models';

interface BookCardProps {
  book: Book;
  currency: string;
  action: () => void;
  actionString: string;
  className: string;
}
const BookCard = ({
  book,
  currency,
  action,
  actionString,
  className
}: BookCardProps) => {
  return (
    <Card style={{ width: '18rem' }} className={className}>
      <Card.Img variant="top" src={book.cover} />
      <Card.Body>
        <Card.Title> {book.title}</Card.Title>
        <Card.Text>
          {book.price}
          {currency}
        </Card.Text>
        <Button variant="primary" onClick={action}>
          {actionString}
        </Button>
      </Card.Body>
    </Card>
  );
};
export { BookCard };
