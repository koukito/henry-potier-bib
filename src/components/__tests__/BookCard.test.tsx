import React from 'react';
import renderer from 'react-test-renderer';

import { BookCard, BookCardProps } from '../BookCard';

const mockBookData = {
  isbn: 'aaa',
  title: 'MockTitlte',
  synopsis: [],
  price: 11,
  cover: ''
};
describe('BookCard', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <BookCard
          book={mockBookData}
          action={() => {}}
          actionString={'mock action string'}
          currency={'â‚¬'}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
