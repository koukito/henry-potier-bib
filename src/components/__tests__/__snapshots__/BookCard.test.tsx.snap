// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookCard renders correctly 1`] = `
<div
  className="card"
  style={
    Object {
      "width": "18rem",
    }
  }
>
  <img
    className="card-img-top"
    src=""
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
       
      MockTitlte
    </div>
    <p
      className="card-text"
    >
      11
      â‚¬
    </p>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      mock action string
    </button>
  </div>
</div>
`;
